src/test/kotlin/se/uulm/snowballr/backend/repository/PaperTableRepoTest.kt

```diff
@@ -83,6 +85,86 @@ class PaperTableRepoTest : RepositoryTest(arrayOf(PaperTable), false) {
         }
     }
 
+    @Nested
+    inner class DoesPaperExistByExternalId {
+        @Test
+        fun `When a paper with the given external id exists, then true is returned`() = runTest {
+            val externalId = "ExternalId"
+            insertPaperAndGetId(externalId = externalId)
+
+            val isPaperExistent = repo.doesPaperExistByExternalId(externalId)
+
+            assertTrue(isPaperExistent)
+        }
+
+        @Test
+        fun `When a paper with the given external id does not exist, then false returned`() = runTest {
+            val externalId = "NonExistentExternalId"
+
+            val isPaperExistent = repo.doesPaperExistByExternalId(externalId)
+
+            assertFalse(isPaperExistent)
+        }
+    }
+
+    @Nested
+    inner class CreatePaper {
+        fun getExamplePaperRequest(externalId: String = "ExternalId"): Paper = Paper.newBuilder()
+            .setExternalId(externalId)
+            .setTitle("Title")
+            .setAbstrakt("Abstract")
+            .setYear(2025)
+            .setPublisher("Publisher")
+            .setPublicationName("PublicationName")
+            .setPublicationType("PublicationType")
+            .build()
+
+        @Test
+        fun `When a paper is created, then the created paper is returned`() = runTest {
+            val request = getExamplePaperRequest()
+
+            val start = OffsetDateTime.now()
+            val createdPaper = repo.createPaper(request)
+            val end = OffsetDateTime.now()
+
+            with(createdPaper) {
+                assertThat(id).isNotNull
```

Comment:
First, you forgot to add `()` at the end, but:

```suggestion
                assertNotNull(id)
```

would be better.

Reply:
Changed it.